Mathlib/NumberTheory/NumberField/Completion.lean

```diff
@@ -0,0 +1,339 @@
+/-
+Copyright (c) 2024 Salvatore Mercuri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Salvatore Mercuri
+-/
+import Mathlib.Algebra.Field.Subfield
+import Mathlib.Analysis.Normed.Module.Completion
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.NumberTheory.NumberField.Embeddings
+import Mathlib.Topology.UniformSpace.Basic
+import Mathlib.Topology.Instances.Real
+
+/-!
+# The completion of a number field at an infinite place
+
+This file contains the completion of a number field at an infinite place.
+
+This is ultimately achieved by applying the `UniformSpace.Completion` functor, however each
+infinite place defines its own `UniformSpace` instance, so the inference system cannot
+automatically infer these. A common approach to handle the ambiguity that arises from having
+multiple sources of instances is through the use of type synonyms. In this case, we define a
+type synonym `WithAbs` for a semiring which depends on an absolute value. This provides a
+systematic way of assigning and inferring instances of the semiring that depend on an absolute
+value. In our application, relevant instances and the completion of a number field `K` are first
+defined at the level of `AbsoluteValue` by using the type synonym `WithAbs` of `K`, and then
+derived downstream for `InfinitePlace` (which is a subtype of `AbsoluteValue`).
+
+## Main definitions
+ - `WithAbs` : type synonym equipping a semiring with an absolute value.
```

Comment:
Thanks! I agree the "equipping" here does not really capture the idea behind using the type synonym. I have expanded here. 

I think I'm not calling it `WithAbsRing` (or I guess that should really be `WithAbsSemiring`) because `Semiring` is required by `AbsoluteValue` and since this synonym aliases the lowest required class for this dependent synonym it seems unambiguous to me -- also see [`WithIdeal`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Nonarchimedean/AdicTopology.html#WithIdeal) which is not called `WithIdealRing` or `WithIdealCommRing` (although this is technically a class rather than a synonym, so I'm not sure if the same naming convention applies). If we were aliasing a `Field` or something else I think I would agree with you on the naming.